compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/DefaultArgumentStubGenerator.kt

```diff
@@ -84,6 +83,23 @@ open class DefaultArgumentStubGenerator(
                 variables[it] = newIrFunction.extensionReceiverParameter!!
             }
 
+            // In order to deal with forward references in default value lambdas,
```

Comment:
An easier way to achieve the same effect is to do

```kotlin
irFunction.valueParameters.associateWithTo(variables) {
    newIrFunction.valueParameters[it.index]
}
```

so that reads of `f1` before the temporary is created are redirected to the original argument.

Reply:
That is a great point and that is indeed much simpler. Thanks!